### ISSUE TITLE
Multiprocessing multiple add_searches 

### ISSUE THREAD

**Issue by mlittmanabbvie** [SimonBlanke/Hyperactive#40]
Hello, it wasn't entirely clear in the multiprocessing example but do the multiple add_searches run in parallel or when would you increase n_jobs? For example, if I want to run two optimizers in parallel (say the repulsive hill climber and the bayesian optimizer) how would I run both in parallel? Would it be like this below?

```python
h = Hyperactive(["progress_bar", "print_results", "print_times"])
search_space = {
    "exp": list(range(0, 5)),
    "slope": list(np.arange(0.001, 10, step=0.05)),
    "freq_mult": list(np.arange(1, 2.5, 0.005)),
    "clust": [5],
    "df": [ret_df],
    "finp": [finp],
    "asc": [False],
    "use_pca": [False],
    "last": [False],
    "disc_type": ["type"],
}
h.add_search(
    func_minl,
    search_space=search_space,
    n_iter=10,
    optimizer=RepulsingHillClimbingOptimizer(
        epsilon=0.05,
        distribution="normal",
        n_neighbours=3,
        rand_rest_p=0.03,
        repulsion_factor=3,
    ),
    n_jobs=1,
    max_score=None,
    initialize={
        "warm_start": [
            {
                "exp": 2,
                "slope": 5,
                "freq_mult": 1.5,
                "clust": 5,
                "df": ret_df,
                "finp": finp,
                "asc": False,
                "use_pca": False,
                "last": False,
                "disc_type": "type",
            }
        ]
    },
    early_stopp={"tol_rel": 1, "n_iter_no_change": 3},
    random_state=0,
    memory=True,
    memory_warm_start=None,
)

h.add_search(
    func_minl,
    search_space=search_space,
    n_iter=maxiter,
    optimizer=BayesianOptimizer(rand_rest_p=0.03, xi=0.03, warm_start_smbo=None),
    n_jobs=1,
    max_score=None,
    early_stopping={"tol_rel": 0.005, "n_iter_no_change": 10},
    random_state=0,
    memory=False,
    memory_warm_start=None,
)

```

**Comment by SimonBlanke** [comment]
Hello @mlittmanabbvie,

yes this is how you would run those two optimization algorithms in parallel. They both run independed of each other and optimize the `func_minl`-function with all the parameters passed to add_search().

If you want to run the same optimization in parallel you could just call `add_search()` multiple times, but that would be tedious to write. Instead you can just call `add_search()` a single time and set `n_jobs` to the number of searches you want to run in parallel. So `n_jobs` is a shortcut.

In the following example we would have 8 parallel searches:
```python
h = Hyperactive()
h.add_search(...)
h.add_search(..., n_jobs=2)
h.add_search(..., n_jobs=4)
h.add_search(...)
h.run()
```

> Hello, it wasn't entirely clear in the multiprocessing example

Could you point me to this example, so that I can add some more explanations to it? This way future users will benefit from it.

**Comment by mlittmanabbvie** [comment]
https://github.com/SimonBlanke/Hyperactive/blob/master/examples/tested_and_supported_packages/multiprocessing_example.py ,

Looking back on the example, you do mention that. I guess I didn't read close enough for whether they were all parallel. If they all run in parallel, is there a resolving function at the end that chooses the best run?

**Comment by SimonBlanke** [comment]
Yes you can access the best parameter set via `.best_para(objective_function)` (from the [API ref](https://github.com/SimonBlanke/Hyperactive#hyperactive-api-reference)). This automatically selects the best parameters from all jobs of the passed objective function.

If you have more questions that fit the topic you can ask them here, otherwise you can close this issue. If you have other questions you can take a look at this [extensive tutorial](https://nbviewer.org/github/SimonBlanke/hyperactive-tutorial/blob/main/notebooks/hyperactive_tutorial.ipynb) or open another issue.

**Comment by mlittmanabbvie** [comment]
You have been so helpful thank you so much!
